### PR TITLE
SLLS-146 Switch issue sync to the new backend

### PR DESCRIPTION
**Pull request by kirill-knize-sonarsource**

### CODE REVIEW

**Comment by kirill-knize-sonarsource** [comment]
Advices on coverage improvement are very welcome.

File: src/test/java/org/sonarsource/sonarlint/ls/clientapi/SonarLintVSCodeClientTests.java [hunk 1]
```diff
@@ -191,19 +195,15 @@ void shouldDoNothingOnShowSmartNotificationWhenConnectionIsNotFound() {
   }
 
   @Test
-  void shouldThrowForStartProgress() {
-    assertThrows(UnsupportedOperationException.class, () -> underTest.startProgress(mock(StartProgressParams.class)));
+  void shouldThrowForStartProgress() throws ExecutionException, InterruptedException {
```
Comment (jblievremont) [inline src/test/java/org/sonarsource/sonarlint/ls/clientapi/SonarLintVSCodeClientTests.java:198]: IMHO this method should be renamed to reflect the new behavior

File: src/test/java/org/sonarsource/sonarlint/ls/mediumtests/ConnectedModeMediumTests.java [hunk 2]
```diff
@@ -382,34 +382,6 @@ void shouldThrowExceptionForUnknownConnection() {
...
-//    var serverUrl = mockWebServerExtension.url("");
-//    var cleanUrl = stripTrailingSlash(serverUrl);
-//    var params = new SonarLintExtendedLanguageServer.GenerateTokenParams(cleanUrl);
-//
-//    var result = lsProxy.generateToken(params);
-//    var actual = result.get();
-//
-//    assertThat(actual.getServerUrl()).isEqualTo(cleanUrl + "/account/security");
-//  }
-
```
Comment (jblievremont) [inline src/test/java/org/sonarsource/sonarlint/ls/mediumtests/ConnectedModeMediumTests.java:412]: 🔥

**Review by jblievremont — APPROVED** [review]
LGTM with a small comment 👍🏻

**Pull request merged**